Doc update (#547)
change-metrics/monocle: README.md

  only exposed in the development version of the docker-compose
  (`docker-compose.yml.dev`).

### Monocle configuration file

The Monocle configuration file defines workspaces. The file is used by the API and crawlers processes. The format of the file is YAML. You might want to use Dhall to manage it or to better
understand the schema ([dhall-monocle](https://github.com/change-metrics/dhall-monocle.)).

#### Workpaces

> - permits to define multiple crawlers - [details](#crawlers)
> - permits to define a list of projects - [details](#projects-definition)
> - permits to define identities - [details](#groups-definition)
> - uses a dedicated ElasticSearch index

Reviewer: How about:
```suggestion
A workspace uses a dedicated ElasticSearch. A workspace defines:
- crawlers - [details](#crawlers)
- a list of projects - [details](#projects-definition)
- identities - [details](#groups-definition)
```

#### Crawlers

Monocle provides two kinds of crawlers:

- Change: A crawler to fetch Changes proposed to a repository. Monocle supports Gerrit (Reviews), GitHub (Pull-Requests), GitLab (Merge-Requests).
- TaskData: A crawler to fetch task data related to a repository. Monocle supports GitHub (issues),
  and BugZilla (Bugs).

> The `.secrets` file is used to store credentials or API keys used by crawlers to authenticate on providers. The configuration file do not contain any sensible information then it might be convenient to store the file on a DVCS.

Reviewer: ```suggestion
The `.secrets` file is used to store credentials and API keys used by crawlers to authenticate on providers.
```

`gerrit_prefix` might be set to configure the crawler to prepend the repository name with a prefix.

Note that this crawler is managed by the `crawler-legacy` container.

###### GitLab

A GitLab provider settings

```YAML
  provider:
    gitlab_organization: redhat/centos-stream/ci-cd/zuul
    # Optional settings
    gitlab_repositories:
      - jobs-config
    gitlab_url: https://gitlab.com/api/graphql
    gitlab_token: GITLAB_TOKEN
```

`gitlab_organization` is the only mandatory key. If `gitlab_repositories` is not specified then
the crawler will crawl the whole organization repositories. If specified then it will crawl only
the listed repositories.

`gitlab_url` might be specified in case of an alternate url. Default is "https://gitlab.com/api/graphql".

`gitlab_token` might be specified to use an alternate environment variable name to look for the
token value. Default is "GITLAB_TOKEN"

Note that this crawler is managed by the `crawler` container.

##### TaskData

Monocle provides additional crawlers to attach tasks/issues/RFEs to changes based on a
match on `change_url`. Then, Changes can be enhanced with information about related
tasks such as a `priority` or a `score`.

> However these crawlers are not fully integrated with the configuration file and the crawler
> container process. Until they are, usage might be a bit rough.

Reviewer: Perhaps we don't have to integrate this, it's good to have a generic task-data interface that can be used freely. In a future version the github and gitlab provider will do this automatically, and there will be a dedicated bugzilla crawler, and we can still keep that generic provider.

```suggestion
Here are some example process you can run to collect task data: 
```

A contributor id on a Gerrit instance is formated as `<domain>/<Full Name>/<gerrit-user-id>`.

### Groups definition

> A group in Monocle permits to group authors of Changes. The Monocle Query language provides facilities
> to filter based on group names.

Reviewer: ```suggestion
A group in Monocle permits to group authors of Changes and filter them from the web interface.
```

Group memberships are defined through the `idents` section of the configuration.

Here is an example:

```YAML
workspaces:
  - name: example
    idents:
      - ident: John Doe
        aliases:
          - github.com/john-doe
          - review.opendev.org/John Doe/12345
        groups:
          - devs
      - ident: Jane Doe
        aliases:
          - github.com/jane-doe
        groups:
          - devs
          - ptl
```

#### Apply idents configuration

Database objects must be updated to reflect the configuration. Once `config.yaml` is updated, run the following commands:

```bash
docker-compose restart crawler-legacy

> docker-compose restart crawler

Reviewer: nit: we shouldn't have to restart the new crawler.
```suggestion
```
